src/game/behaviors/hidden_star.inc.c

```diff
@@ -47,12 +47,19 @@ void bhv_hidden_star_trigger_loop(void) {
     }
 }
 
+void bhv_bowser_course_red_coin_star_init(void) {
+    s16 numRedCoinsRemaining = count_objects_with_behavior(bhvRedCoin);
+
+    o->oHiddenStarTriggerTotal = numRedCoinsRemaining + gRedCoinsCollected;
```

Comment:
Actually, this implementation does work for multiple red coin stars across different areas. The only thing that's "inaccurate" is the red coin count, since it's shared across areas. The red coin stars still spawn when all the red coins in their respective area have been collected. Multi-area red coin challenges still would require a big rewrite of the red coin system

Reply:
I think there's a simple solution here actually, and that's to use bparam3 or bparam4 as a means of overriding absolute coin count. This would be useful in cases where coin count can't be determined automatically (i.e. coins in different areas).

Reply:
If you do add this, leave a comment somewhere describing it so that it doesn't just turn into a trade secret lol

Reply:
where would be a good place to leave it though? in practice you'll never look for star code when making red coin star missions so idk how people would really know this exists

Reply:
Really the correct answer would be the Wiki, but... 🤷

At this point we're in extremely heavy Wiki debt, so probably just the README so that it ends up _somewhere_ before entering the knowledge void. I don't think obscurity is a good enough reason to say we can't ever implement a feature, especially in this case since the feature doesn't actively prevent red coins from working the way they normally do.

Reply:
The only reason it would is if somebody tried to repurpose one of the bparams for something else, at which point if the use is commented in the code, then that base is covered at least.